### PR TITLE
[B2BP-178] Implementing PreHeader as generic component

### PR DESCRIPTION
**Pull request by Meolocious**
#### List of Changes
1. Modified the "layout.tsx" file to implement the PreHeader component from @pagopa/pagopa-editorial-components.
2. Added a new file named "preHeaderData.json" for testing the data used in the PreHeader component.

#### Motivation and Context
1. The changes to the "layout.tsx" file were required to incorporate the PreHeader component into the project. This component is essential for the layout and navigation of the website.
2. The "preHeaderData.json" file was introduced to facilitate testing and validation of data used in the PreHeader component. Having realistic data helps ensure that the PreHeader component functions as expected and aligns with the design and content requirements.

#### How Has This Been Tested?
1. The modified "layout.tsx" file was tested in a development environment to verify that the PreHeader component integrated correctly and that it displays the expected content and layout.
2. The "preHeaderData.json" file was used to supply data to the PreHeader component during testing. This allowed us to validate that the component handles real data effectively.

#### Types of changes
- [ ] Chore (nothing changes from a user perspective)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Doesn't need any documentation updates

### CODE REVIEW

**Comment by Meolocious** [comment]
There's some errors to be checked for the component. Inside the file "layout.tsx" you'll find twice the PreHeader component:

1) First is the component directly from the storybook, copied\pasted the exact component.
2) Second is the same component but extrapolated datas from the first one into a json file, same exact copy of them.

This was necessary to let also both you check the second version of the PreHeader component, the one that technically should work in the project, with datas that we'll change to an API call. This component, from my IDE, have always type errors even if the datas are the same and even if I'll try to dig deeper and deeper declaring types or intercept them, at least there will be always some errors of type (or something missing while before it was not missing at all). I digged also inside the main repo of @pagopa/pagopa-editorial-components to search the structure and match it but nothing changed.

For now the eslint is always in error due to some complication to fix the errors. This is a draft PR to check other things before any check on eslint part.

**Comment by Meolocious** [comment]
IMPORTANT NOTES: before this PR is better to approve and merge the other one https://github.com/pagopa/b2b-portals/pull/39

File: apps/nextjs-website/src/lib/api/preHeaderAPI.ts
Comment (datalek) [inline apps/nextjs-website/src/lib/api/preHeaderAPI.ts:1]: Can you add test for `getPreHeader` method?

File: apps/nextjs-website/src/app/[...slug]/page.tsx [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
+import { Page } from '@/lib/api/navigation/pages';
 
-// Dynamic segments not included in generateStaticParams will return a 404.
-// more: https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#dynamicparams
 export const dynamicParams = false;
-
-// Statically generate routes at build time instead of on-demand at request time.
-// more: https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#generating-static-params
-export const generateStaticParams = () => getAllPages();
+export const generateStaticParams = () => getAllPages() as Promise<Page[]>;
```
Comment (kin0992) [inline apps/nextjs-website/src/app/[...slug]/page.tsx:5]:
```suggestion
export const generateStaticParams = () => getAllPages();
```

File: apps/nextjs-website/src/components/PreHeaderClient.tsx
Comment (kin0992) [inline apps/nextjs-website/src/components/PreHeaderClient.tsx:1]: I'd call this component just `PreHeader`, without adding the `Client` suffix

File: package-lock.json
Comment (kin0992) [inline package-lock.json:1]: why there are changes on the lock-file?

Comment (Meolocious) [reply package-lock.json:1]: no sense changes. Got several errors on the lock-file after I deleted node_modules and reinstalled again everything. The branch is alligned to the main so I really don't know. Seems now to be fixed

File: apps/nextjs-website/src/components/PreHeaderClient.tsx
Comment (Meolocious) [reply apps/nextjs-website/src/components/PreHeaderClient.tsx:1]: done

File: apps/nextjs-website/src/app/[...slug]/page.tsx (hunk 1 above)
Comment (Meolocious) [reply apps/nextjs-website/src/app/[...slug]/page.tsx]: done

**Pull request closed without merging**